### DISCUSSION TITLE
moveIndex

### DISCUSSION THREAD

**Discussion by cjquines** [remeda/remeda#954] [Suggest a Function]
Equivalent to [`move` from Ramda](https://ramdajs.com/docs/#move); the [most requested function in Lodash](https://github.com/lodash/lodash/issues?q=sort%3Areactions-%2B1-desc+).

The one-liner `R.moveIndex(data, from, to)` would be `data.toSpliced(from, 1).toSpliced(to, 0, data[from])`, but this would also have `R.moveIndex.strict`, like `R.swapIndices.strict`, which is another reason to have this function in Remeda proper.

**Comment by eranhirsch** [comment]
I'm cool with a `moveItem` function.

> but this would also have R.moveIndex.strict, like R.swapIndices.strict, which is another reason to have this function in Remeda proper.

I couldn't find a `swapIndices.strict` in Remeda, what did you mean by this?

**Comment by cjquines** [comment]
oops, i thought we had a swapIndices.strict. i was referring to the fact that swapIndices was typed for tuples:

```ts
expectTypeOf(swapIndices([2, 'a'], 0, 1)).toEqualTypeOf<[string, number]>();
```

a hypothetical `moveItem` should also be like this:

```ts
expectTypeOf(moveItem([2, 'a'], 0, 1)).toEqualTypeOf<[string, number]>();
```

**Comment by eranhirsch** [comment]
Yea, it should provide good typing for tuples
